### PR TITLE
Use the neo4j_database parameter everywhere

### PR DESCRIPTION
**Pull request by stellasia**
# Description

Fix bug reported in #214 

In addition:
- Add `neo4j_database` parameter where it was missing (`Text2CypherRetriever` and `Neo4jChunkReader`)
- Update all examples to use the recommended way to set the db (ie not using the driver's database parameter which is undocumented)
- Expose the `neo4j_database` parameter in `SimpleKGPipeline` since this is the only way to configure it properly.

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

Complexity: Low

## How Has This Been Tested?
- [x] Unit tests
- [ ] E2E tests
- [x] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [x] Unit tests have been updated
- [ ] E2E tests have been updated
- [x] Examples have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed
- [x] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,8 +3,18 @@
 ## Next
 
 ### Added
-- Introduced optional lexical graph configuration for SimpleKGPipeline, enhancing flexibility in customizing node labels and relationship types in the lexical graph.
+- Introduced optional lexical graph configuration for `SimpleKGPipeline`, enhancing flexibility in customizing node labels and relationship types in the lexical graph.
+- Introduced optional `neo4j_database` parameter for `SimpleKGPipeline` to create the graph in a non-default database.
+- Introduced optional `neo4j_database` parameter for `Neo4jChunkReader` to read chunks from a non-default database.
+- Introduced optional `neo4j_database` parameter for `Text2CypherRetriever` to run the generated query against a non-default database.
 - Ability to provide description and list of properties for entities and relations in the SimpleKGPipeline constructor.
+- Ability to run queries against a custom `neo4j_database` for the `Text2CypherRetriever` and the `Neo4jChunkReader` component (was already possible with other retrievers and components).
```
Comment (alexthomas93) [inline CHANGELOG.md:11]: It's a bit nit-picky but I feel like there's a bit of redundancy in this change log update. Maybe combine a few of these lines?

**Review by alexthomas93 — APPROVED** [review]
LGTM. Just had a small comment about the changelog

File: examples/build_graph/simple_kg_builder_from_text.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 # Neo4j db infos
 URI = "neo4j://localhost:7687"
 AUTH = ("neo4j", "password")
-DATABASE = "neo4j"
+DATABASE = "newdb"
```
Comment (willtai) [inline examples/build_graph/simple_kg_builder_from_text.py:24]: I'm curious, why the change?

**Review by willtai — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (stellasia) [reply CHANGELOG.md]: Agreed, it was a bit weird... Updated in https://github.com/neo4j/neo4j-graphrag-python/pull/216/commits/80718ee38fbda1cf4e147e4c1a50eedfb12e54bc
